src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -151,11 +151,13 @@ string_exprt string_constraint_generatort::add_axioms_from_int_with_radix(
 {
   const typet &type=input_int.type();
   PRECONDITION(type.id()==ID_signedbv);
+  /// Most of the time we can evaluate radix as an integer. The value 0 is used
+  /// to indicate when we can't tell what the radix is.
+  const unsigned long radix_ul=to_integer_or_default(radix, 0);
+  PRECONDITION((radix_ul>=2 && radix_ul<=36) || radix_ul==0);
```

Comment:
It does not seem like a precondition to me, I think `CHECK_RETURN` would be more appropriate.